includes/RestApi/SiteGenController.php

```diff
@@ -105,4 +183,117 @@ public function generate_sitegen_meta( \WP_REST_Request $request ) {
 		// TODO Implement the main function and do computations if required.
 		return SiteGenService::instantiate_site_meta( $site_info, $identifier, $skip_cache );
 	}
+
+	/**
+	 * Gets the preview homepages
+	 *
+	 * @param \WP_REST_Request $request parameter.
+	 * @return array
+	 */
+	public function get_homepages( \WP_REST_Request $request ) {
+
+		$site_description = $request->get_param( 'site_description' );
+		$regenerate       = $request->get_param( 'regenerate' );
+		$site_info        = array( 'site_info' => array( 'site_description' => $site_description ) );
+
+		// If the option exists and is not empty, return it.
+		$existing_homepages = get_option( Options::get_option_name( 'sitegen_homepages' ), array() );
+		if ( ! empty( $existing_homepages ) && ! $regenerate ) {
+			return new \WP_REST_Response( $existing_homepages, 200 );
+		}
+
+		$target_audience = SiteGenService::instantiate_site_meta( $site_info, 'targetaudience', true );
+		$content_style   = SiteGenService::instantiate_site_meta( $site_info, 'contenttones', true );
+
+		if ( ! $target_audience || ! $content_style ) {
+			return new \WP_Error(
+				'nfd_onboarding_error',
+				__( 'Required data is missing.', 'wp-module-onboarding' ),
+				array(
+					'status' => 400,
+				)
+			);
+		}
```

Comment:
What if we add a code to send Error from our Service and catch it here, cause I believe here what if it gives an error from Press 3 API, that won't be null but still have an error and going forward will break things.